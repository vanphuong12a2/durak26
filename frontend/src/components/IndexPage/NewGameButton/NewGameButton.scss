.new-game {
  text-align: center;
}

.error {
  text-align: center;
  color: darkred;
}

.loading {
  text-align: center;
  color: darkblue;
}
