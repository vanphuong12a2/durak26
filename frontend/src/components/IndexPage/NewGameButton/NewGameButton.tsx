import React from 'react';
import {RouteComponentProps} from 'react-router-dom';
import './NewGameButton.scss';

export interface GamesProps {
  gameId?: string
  loading: boolean
  error?: Error
  newGameButtonOnClick: () => any
}

const NewGameButton = (props: GamesProps & RouteComponentProps) => {

  if (props.error) {
    return (<div className='error'>There is an error!</div>)
  }

  if (props.loading) {
    return (<div className='loading'>Loading...</div>)
  }

  if (props.gameId) {
    props.history.push(`/play/${props.gameId}`);
  }

  return (
    <div className='new-game'>
      <a className='waves-effect waves-light btn-small' onClick={props.newGameButtonOnClick}>New game</a>
    </div>
  );
};

export default NewGameButton;
