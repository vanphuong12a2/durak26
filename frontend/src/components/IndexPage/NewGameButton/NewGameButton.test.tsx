import React from 'react';
import {create} from 'react-test-renderer';
import NewGameButton from './NewGameButton';
import createMemoryHistory from 'history/createMemoryHistory';
import {shallow} from 'enzyme';

describe('<NewGameButton />', () => {

  const history = createMemoryHistory(undefined);
  const match = {params: [], isExact: false, path: '', url: ''};

  describe('Snapshots', () => {
    it('renders new game button correctly', () => {
      const component = create(
        <NewGameButton
          loading={false}
          newGameButtonOnClick={jest.fn()}
          history={history}
          location={history.location}
          match={match}
        />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders loading game correctly', () => {
      const component = create(<NewGameButton
        loading={true}
        newGameButtonOnClick={jest.fn()}
        history={history}
        location={history.location}
        match={match}
      />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders error correctly', () => {
      const component = create(<NewGameButton
        loading={true}
        error={new Error('someError')}
        newGameButtonOnClick={jest.fn()}
        history={history}
        location={history.location}
        match={match}
      />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  it('should redirect to game page if gameId is available', () => {
    shallow(
      <NewGameButton
        gameId={123}
        loading={false}
        newGameButtonOnClick={jest.fn()}
        history={history}
        location={history.location}
        match={match}
      />);

    expect(history.location.pathname).toEqual('/play/123')
  });
});
