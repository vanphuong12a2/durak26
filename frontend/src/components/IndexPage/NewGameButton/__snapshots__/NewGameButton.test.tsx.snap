// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewGameButton /> Snapshots renders error correctly 1`] = `
<div
  className="error"
>
  There is an error!
</div>
`;

exports[`<NewGameButton /> Snapshots renders loading game correctly 1`] = `
<div
  className="loading"
>
  Loading...
</div>
`;

exports[`<NewGameButton /> Snapshots renders new game button correctly 1`] = `
<div
  className="new-game"
>
  <a
    className="waves-effect waves-light btn-small"
    id="new-game-btn"
    onClick={[MockFunction]}
  >
    New game
  </a>
</div>
`;
